modules/python/python-logmsg.c

```diff
@@ -270,6 +274,83 @@ py_log_message_set_pri(PyLogMessage *self, PyObject *args, PyObject *kwrds)
   Py_RETURN_NONE;
 }
 
+static PyObject *
+_datetime_timestamp(PyObject *py_datetime)
+{
+  PyObject *py_tzinfo = _py_get_attr_or_null(py_datetime, "tzinfo");
+  if (!py_tzinfo)
+    {
+      PyErr_Format(PyExc_ValueError, "Error obtaining tzinfo");
+      return NULL;
+    }
+  PyObject *py_epoch = PyDateTimeAPI->DateTime_FromDateAndTime(1970, 1, 1, 0, 0, 0, 0, py_tzinfo,
+                       PyDateTimeAPI->DateTimeType);
+
+  PyObject *py_delta = _py_invoke_method_by_name(py_datetime, "__sub__", py_epoch,
+                                                 "PyDateTime", "py_datetime_to_logstamp");
+  if (!py_delta)
+    {
+      Py_XDECREF(py_tzinfo);
+      Py_XDECREF(py_epoch);
+      PyErr_Format(PyExc_ValueError, "Error calculating POSIX timestamp");
+      return NULL;
+    }
+
+  PyObject *py_posix_timestamp = _py_invoke_method_by_name(py_delta, "total_seconds", NULL,
+                                                           "PyDateTime", "py_datetime_to_logstamp");
+  Py_XDECREF(py_tzinfo);
+  Py_XDECREF(py_delta);
+  Py_XDECREF(py_epoch);
+
+  return py_posix_timestamp;
+}
+
+static gint
+_datetime_get_gmtoff(PyObject *py_datetime)
+{
+  PyObject *py_utcoffset = _py_invoke_method_by_name(py_datetime, "utcoffset", NULL, "PyDateTime",
+                                                     "py_datetime_to_logstamp");
+  if (!py_utcoffset)
+    {
+      PyErr_Format(PyExc_ValueError, "Error obtaining timezone info");
+      return -1;
+    }
+
+  if (py_utcoffset == Py_None)
```

Comment:
Even in case of `Py_None`, py_utcoffset must be decrefed: https://docs.python.org/3/c-api/none.html#c.Py_None.

Reply:
Fixed.